pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -64,6 +63,45 @@ def _are_inverses(op1, op2):
     return False
 
 
+def _can_cancel_ops(op1, op2):
```

Comment:
I'm a little confused about the differences in concepts between `_are_inverses` and `_can_cancel_ops`. Can we combine them together into one function? This separation kinda implies we can cancel them when they are not inverses, or they could be inverses but we can't cancel them.  I think there's a one-to-one mapping between being an inverse and being cancellable.